alvr/client_opengl/cpp/graphics.cpp

```diff
@@ -660,9 +670,13 @@ void renderEye(
     GL(glUseProgram(0));
 }
 
-void ovrRenderer_RenderFrame(ovrRenderer *renderer, const EyeInput input[2], bool isLobby) {
-    if (renderer->enableFFR) {
-        renderer->ffr->Render();
+void ovrRenderer_RenderFrame(ovrRenderer *renderer, const EyeInput input[2], bool isLobby, bool isReprojected) {
+    if (!isLobby && !isReprojected) {
+        if (renderer->enableFFR) {
+          renderer->ffr->Render();
+        } else {
+          renderer->staging_pass->Render();
+        }
```

Comment:
Always run `renderer->staging_pass->Render();` first if `!isLobby` and pipe the output to FFR. You can modify ffr.cpp so it never has to deal with OES textures. You could also run the staging pass even when `isLobby`. This should simplify the code a bit.